client_request_metrics_test.py

```diff
@@ -0,0 +1,784 @@
+        cluster.populate(2, debug=True)
+        self.start_cluster()
+        node1 = cluster.nodelist()[0]
+
+        global jmx
+        jmx = JolokiaAgent(node1)
+        jmx.start()
+
+        s = self.session = self.patient_exclusive_cql_connection(node1, retry_policy=FallthroughRetryPolicy(), request_timeout=30)
+        for k in [KEYSPACE, FAIL_WRITE_KEYSPACE]:
+            s.execute(f"CREATE KEYSPACE {k} WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': '2'}}")
+            s.execute(f"CREATE TABLE {k}.{TABLE} (k int, c int, v int, PRIMARY KEY (k,c))")
+
+        s.execute(f"CREATE KEYSPACE {VIEW_KEYSPACE} WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': '1'}}")
+        s.execute(f"CREATE TABLE {VIEW_KEYSPACE}.{TABLE} (k int, c int, v int, PRIMARY KEY (k,c))")
+        s.execute(f"CREATE MATERIALIZED VIEW {VIEW_KEYSPACE}.{VIEW} AS SELECT * FROM {TABLE} WHERE c IS NOT NULL AND k IS NOT NULL PRIMARY KEY (c,k);")
+
+        for c in range(TOMBSTONE_FAILURE_THRESHOLD + 1):
+            self.session.execute(f"DELETE FROM {KEYSPACE}.{TABLE} WHERE k={TOMBSTONE_FAIL_KEY} AND c={c}")
+
+        node1.watch_log_for("Created default superuser role 'cassandra'")  # don't race with async default role creation, which creates a write
+        node1.watch_log_for('Completed submission of build tasks for any materialized views defined at startup', filename='debug.log')  # view builds cause background reads
+
+    def start_cluster(self):
+        self.cluster.start(jvm_args=[f"-Dcassandra.test.fail_writes_ks={FAIL_WRITE_KEYSPACE}"])
+
+    def test_client_request_metrics(self):
+        self.setup_once()
+
+        # DecayingEstimatedHistogramReservoir
+        self.write_nominal()
+        self.read_nominal()
+
+        self.write_failures()
+        self.write_unavailables()
+        self.write_timeouts()
+
+        self.read_failures()
+        self.read_unavailables()
+        self.read_timeouts()
+
+        self.range_slice_failures()
+        self.range_slice_unavailables()
+        self.range_slice_timeouts()
+
+        self.view_writes()
+
+        self.cas_read()
+        self.cas_read_contention()
+        self.cas_read_failures()
+        self.cas_read_unavailables()
+        self.cas_read_timeouts()
+
+        self.cas_write()
+        self.cas_write_contention()
+        self.cas_write_unavailables()
+        self.cas_write_timeouts()
+        self.cas_write_condition_not_met()
+
+    def write_nominal(self):
+        query_count = 5
+        global_diff, cl_diff = self.validate_nominal('Write',
+                                                     WriteMetrics,
+                                                     SimpleStatement(f"INSERT INTO {KEYSPACE}.{TABLE} (k,c) VALUES (0,0)", consistency_level=CL.ONE),
+                                                     query_count)
+
+        assert global_diff['MutationSizeHistogram.Count'] == query_count
```

Comment:
We could easily test all valid consistency levels here, instead on only `CL.ONE`:
```suggestion
        query = f"INSERT INTO {KEYSPACE}.{TABLE} (k,c) VALUES (0,0)"
        for cl in (CL.LOCAL_ONE, CL.ONE, CL.TWO, CL.ALL, CL.ANY, CL.LOCAL_QUORUM):
            global_diff, cl_diff = self.validate_nominal('Write',
                                                         WriteMetrics,
                                                         SimpleStatement(query, consistency_level=cl),
                                                         query_count)
```